core/src/main/scala/org/apache/spark/shuffle/ShuffleBlockPusher.scala

```diff
@@ -463,7 +463,7 @@ private[spark] object ShuffleBlockPusher {
 
   private val BLOCK_PUSHER_POOL: ExecutorService = {
     val conf = SparkEnv.get.conf
-    if (Utils.isPushBasedShuffleEnabled(conf)) {
+    if (Utils.isPushBasedShuffleEnabled(conf, isDriver = false)) {
```

Comment:
We can probably change this to `, isDriver = SparkContext.DRIVER_IDENTIFIER == SparkEnv.get.executorId` to be forward compatible

Reply:
Nice call, fixed.